### PR TITLE
UBER-772: require having employee mixin to allow Staff mixin

### PR DESCRIPTION
**Pull request by ThetaDR**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues
https://front.hc.engineering/workbench/platform/tracker/UBER-772
A list of closed updated issues

### CODE REVIEW

File: plugins/view-resources/package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "@hcengineering/setting": "^0.6.9",
     "@hcengineering/text-editor": "^0.6.0",
     "@hcengineering/query": "^0.6.7",
+    "@hcengineering/hr": "^0.6.10",
```
Comment (haiodo) [inline plugins/view-resources/package.json]: Nope, not allowed.

File: plugins/view-resources/src/components/EditDoc.svelte [hunk 2]
```diff
@@ -104,7 +105,10 @@
         m.kind === ClassifierKind.MIXIN &&
         !ignoreMixins.has(m._id) &&
         (hierarchy.hasMixin(object, m._id) ||
-          (showAllMixins && hierarchy.isDerived(realObjectClass, hierarchy.getBaseClass(m._id))))
+          (showAllMixins &&
```
Comment (haiodo) [inline plugins/view-resources/src/components/EditDoc.svelte:107]: Base class in our case is Person, not Employee, so we need to filter all mixins added to class extends our base class.

Comment (haiodo) [reply plugins/view-resources/src/components/EditDoc.svelte:107]: Ouch, Employee is mixin for now, we need to check for child mixins, if document is already had it, and only in this case we should allow child mixins.

Comment (ThetaDR) [reply plugins/view-resources/src/components/EditDoc.svelte:107]: Should be fixed now

**Review by haiodo — APPROVED** [review]

**Pull request merged**